chain-connect/src/helpers.ts

```diff
@@ -12,8 +12,22 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+import { signatures } from "@gala-chain/api";
 import { Eip1193Provider } from "ethers";
 
+export function calculatePersonalSignPrefix(payload: object): string {
+  const payloadLength = signatures.getPayloadToSign(payload).length;
+  const prefix = "\u0019Ethereum Signed Message:\n" + payloadLength;
+
+  const newPayload = { ...payload, prefix };
+  const newPayloadLength = signatures.getPayloadToSign(newPayload).length;
+
+  if (payloadLength === newPayloadLength) {
+    return prefix;
+  }
+  return calculatePersonalSignPrefix(newPayload);
+}
```

Comment:
It can be done without recursion, which might reduce the call somewhat, but because of weird changes where we end up with an extra digit (for example length of 99 that would turn into length of 100 and that would increase the length)

I changed it to only calculate the payload once in a loop though